tests/test_pulp_handler.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import os
+
+import mock
+from pubtools.pluggy import pm  # pylint: disable=wrong-import-order
+from six.moves.urllib.parse import urljoin
+
+from pubtools.exodus.hooks import pulp
+
+MOCK_ENV = {
+    "EXODUS_ENABLED": "True",
+    "EXODUS_GW_URL": "https://exodus-gw.example.com",
+    "EXODUS_GW_ENV": "test",
+    "EXODUS_GW_CERT": "path/to/ex.crt",
+    "EXODUS_GW_KEY": "path/to/ex.key",
+}
+
+
+@mock.patch.dict(os.environ, MOCK_ENV)
+def test_exodus_pulp_hanlder(requests_mock, caplog):
+    caplog.set_level(logging.DEBUG, "pubtools-exodus")
+
+    # Construct handler
+    auth_resp = {
+        "client": {
+            "roles": [],
+            "authenticated": False,
+            "serviceAccountId": "test-client",
+        },
+        "user": {
+            "roles": ["tester"],
+            "authenticated": True,
+            "internalUsername": "test-user",
+        },
+    }
+    auth_url = urljoin(MOCK_ENV["EXODUS_GW_URL"], "/whoami")
+    requests_mock.get(auth_url, json=auth_resp, status_code=200)
+
+    # Simulate task start
```

Comment:
I think you could use "with task_context()" here - per https://release-engineering.github.io/pubtools/hooks.html#pubtools.pluggy.task_context .

It would have the benefits that:

- the task_start/stop hooks are invoked 'for real' by the plugin manager, so you know it really works (here you just invoke them directly so you never really know if they were registered properly)
- task_stop gets invoked (thus unregistering handler) even if the test fails in the middle.  Right now that doesn't happen, leaking a handler which could mess up later tests (once some exist).